### DISCUSSION TITLE
Scripting & Serial Port Access

### DISCUSSION THREAD

**Discussion by jkwim** [arendst/Tasmota#22087] [Q&A]
I am using LD2410B device which is connected to GPIO1/3 HW Serial Port.

The sensor is sending the data more than one time within a second.

When the sensor is working, I can use SerialSend5 to send commands to the sensor and can receive the responses to the command.

So what it means is while the sensor driver is active, SerialSend5 command has access to the serial port.

When I try to access the serial port via script the access to serial port is not possible.

```
>D 48
res=0
tvar=0

tstr=""

>B
print B:
res=so(3 1 256000 8N1 1024)

```

```
14:21:44.008 SCR: warning, pins already used
14:21:44.012 SCR: Serial port set to 8N1 256000 bit/s at rx=3 tx=1 rbu=1024 uart=0
14:21:44.014 SNS: Hardware Serial
```

Is there way to access the already opened serial port via scripting?

My second question is how to read the bytes in the receive buffer and convert them to hex representation 
inside the script?

There is a command to convert a received string in to decimal ie.
`hd("hstr") = converts hex number string to a decimal number
`
what is the hstr here?

Can it be like?

`hstr=sa()`

**Comment by sfromis** [comment]
My expectation would be that you cannot share a serial port (each usage place "owns" it), but instead is has to be either-or.

And as I read that `hd` function, the argument (as indicated by `"`) is simply a string, where you can reasonably expect it to work for a variable, as long as you make sure that it is a string variable. While you *can* name that variable `hstr`, no particular reason to use that name.

**Comment by jkwim** [comment]
> When the sensor is working, I can use SerialSend5 to send commands to the sensor and can receive the responses to the command.
> 
> So what it means is while the sensor driver is active, SerialSend5 command has access to the serial port.

OK. When I looked at the code it appears that SerialSend5 may be sending the command but the received data is accessed and printed by the LD2410 driver itself.

So that explains that there is no contention to read the data by two different functions of the code.

![image](https://github.com/user-attachments/assets/85e0a1b3-63ca-4be6-9155-6fc0cd56681a)

![image](https://github.com/user-attachments/assets/a3951511-9039-4344-8533-737f30d4c7f2)
